mlrun/api/api/utils.py

```diff
@@ -470,6 +471,76 @@ def process_function_service_account(function):
     )
 
 
+def ensure_function_security_context(function, auth_info: mlrun.api.schemas.AuthInfo):
+    """
+    For iguazio we enforce that pods run with user id and group id depending on
+    mlrun.mlconf.function.spec.security_context.enrichment_mode
+    and mlrun.mlconf.function.spec.security_context.enrichment_group_id
+    """
+    function: mlrun.runtimes.pod.KubeResource
+
+    # if security context is not required.
+    # security context is not yet supported with spark runtime since it requires spark 3.2+
+    if (
+        mlrun.mlconf.function.spec.security_context.enrichment_mode
+        == SecurityContextEnrichmentModes.disabled.value
+        or mlrun.runtimes.RuntimeKinds.is_local_runtime(function.kind)
+        or function.kind == mlrun.runtimes.RuntimeKinds.spark
+        or mlrun.mlconf.httpdb.authentication.mode != "iguazio"
+    ):
+        logger.debug(
+            "Security context is not required",
+            mode=mlrun.mlconf.function.spec.security_context.enrichment_mode,
+            function_name=function.metadata.name,
+        )
```

Comment:
I feel like this is going to be a very spammy log no ? (every time you're running a job in the open source/local/spark)